From 7e03816e9c1e651c9d593ebdbcbd3627b7a0ebcd Mon Sep 17 00:00:00 2001
From: Emil Renner Berthing <kernel@esmil.dk>
Date: Mon, 28 Jun 2021 20:54:03 +0200
Subject: [PATCH 53/55] hwmon: (sfctemp) Handle timeouts properly

The wait_for_completion_interruptible_timeout() function returns 0 on
timeout, so return -ETIMEDOUT when that happens.

Signed-off-by: Emil Renner Berthing <kernel@esmil.dk>
---
 drivers/hwmon/sfctemp.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/hwmon/sfctemp.c b/drivers/hwmon/sfctemp.c
index 4328aacf9272..35d3ca682e50 100644
--- a/drivers/hwmon/sfctemp.c
+++ b/drivers/hwmon/sfctemp.c
@@ -134,8 +134,11 @@ static int sfctemp_convert(struct sfctemp *sfctemp, long *val)
 
 	ret = wait_for_completion_interruptible_timeout(&sfctemp->conversion_done,
 							msecs_to_jiffies(10));
-	if (ret < 0)
+	if (ret <= 0) {
+		if (ret == 0)
+			ret = -ETIMEDOUT;
 		goto out;
+	}
 
 	/* calculate temperature in milli Celcius */
 	*val = (long)((readl(sfctemp->regs) & SFCTEMP_DOUT_MSK) >> SFCTEMP_DOUT_POS)
-- 
2.30.2

